core/src/main/java/org/elasticsearch/env/Environment.java

```diff
@@ -108,28 +109,32 @@
     }
 
     public Environment(Settings settings) {
+        this(PathUtils.getDefaultFileSystem(), settings);
+    }
+
+    public Environment(FileSystem fs, Settings settings) {
```

Comment:
> And see my comment above about why java.io.tmpdir should not matter.

Sure, that was a race condition. :)

> But we shouldn't be testing every FS.

That's a red herring, that's not what is being proposed.

> So you can use posix for testing the stuff works, and "basic" for testing it skips the extra code gracefully for non-posix FS.

I'm not sure I agree, I think that we should be testing with configurations that get us close to what most of our users are running on. This increases the likelihood that we detect issues locally before they even hit the CI.